exo-go/src/app_runner/index.go

```diff
@@ -90,26 +91,29 @@ func (a *AppRunner) Start() error {
 	if err != nil {
 		return err
 	}
+	var wg sync.WaitGroup
+	var onlineTextRegex *regexp.Regexp
 	for role, onlineText := range onlineTexts {
-		if err = a.waitForOnlineText(process, role, onlineText); err != nil {
+		wg.Add(1)
+		onlineTextRegex, err = regexp.Compile(fmt.Sprintf("%s.*%s", role, onlineText))
+		if err != nil {
 			return err
 		}
+		go func(role string, onlineTextRegex *regexp.Regexp) {
+			a.waitForOnlineText(process, role, onlineTextRegex)
+			wg.Done()
+		}(role, onlineTextRegex)
 	}
+	wg.Wait()
```

Comment:
We can move back to doing this all at the same time since we no longer have multiple goroutines reading from the same pipe and instead each is adding a new func to be called anytime there is new output